nixos/tests/all-tests.nix

```diff
@@ -21,7 +21,7 @@ let
     if isAttrs val
     then
       if hasAttr "test" val then callTest val
-      else mapAttrs (n: s: discoverTests s) val
+      else mapAttrs (n: s: if n == "passthru" then s else discoverTests s) val
```

Comment:
iirc this code is used by both `release.nix` and `nixosTests`, and it has to deal with all tests, which are still written in a number of styles.
So this is hard to review; can't really tell the implications without a lot of manual testing.

In my opinion `release.nix` should be rewritten to follow the more easily understood structure of `nixosTests`, but until that's done, we need to consider that both code paths used this ill-conceived semi-generic code.

All that aside, we should really try to avoid the proliferation of the *term* "passthru". Unfortunately it's the mechanism we have to use with `mkDerivation`, but it is by itself meaningless, and its presence in the package attribute set is an implementation detail. So if we're going to accept a fix for the problem it would be great for it not to use the `passthru` attribute.

As for rewriting `release.nix`, ideally we'd have something like https://github.com/NixOS/nixpkgs/pull/176557 which has a nice and general solution for this particular problem of overriding a test to use a specific package. But the actual implementation was a bit messy, which is why [I think](https://github.com/NixOS/nixpkgs/pull/176557#issuecomment-1554080211) it should be rewritten. It would probably still resemble the [documented example](https://github.com/NixOS/nixpkgs/pull/176557/files#diff-99f4eb67c3def9d78bbb9ec86a7dced190f46ee7151c77dcbb15a6a08727a330).